peering/peering.go

```diff
@@ -187,6 +187,19 @@ func (ps *PeeringService) Start() error {
 	return nil
 }
 
+//	GetState get the State of the Peering Service
+func (ps *PeeringService) GetState() string {
+	switch ps.state {
```

Comment:
Dont return magic strings pls, this essentially remove type safety.

What you can do instead:
- create and export a custom type for the enum (i belive here you just need to export it)
- create `func (peeringState) String() string` method on that type (wich is the code you wrote here).
- return this custom type

Reply:
To simplify, we can just evaluate the  `ps.state` and return the custom type `PeeringState`

```
type state int
type PeeringState string

const (
	stateInit state = iota
	stateRunning
	stateStopped

	PeerStateInit    PeeringState = "init"
	PeerStateRunning PeeringState = "running"
	PeerStateStopped PeeringState = "stopped"
	PeerStateUnknown PeeringState = "unknown"
)
.....
func (s state) String() PeeringState {
	switch s {
	case stateInit:
		return PeerStateInit
	case stateRunning:
		return PeerStateRunning
	case stateStopped:
		return PeerStateStopped
	}
	return PeerStateUnknown
}

....
func (ps *PeeringService) GetState() state {
	ps.mu.Lock()
	defer ps.mu.Unlock()
	return ps.state
}
```

This way, when an external module uses or wraps the PeeringService, they can get the state using
```
ps.GetState() - returns state (int) or ps.GetState().String() - returns PeeringState of type string.
```

Reply:
Thid is not what i meant, in case that was not clear here you do not have to create a new type juste take `state` and rename it `PeeringState.`

Also `fmt.Stringer` (the string method, must return a string not a custom type)